File: problemas/1000/1061.py
import datetime

diaInicio = int(input().replace('Dia ', ''))
(hInicio, mInicio, sInicio) = map(int, input().replace(' ', '').split(':'))
diaFim = int(input().replace('Dia ', ''))
(hFim, mFim, sFim) = map(int, input().replace(' ', '').split(':'))

ano = datetime.date.today().year
dataInicio = datetime.datetime(ano, 4, diaInicio, hInicio, mInicio, sInicio)
dataFim = datetime.datetime(ano, 4, diaFim, hFim, mFim, sFim)
duracao = dataFim - dataInicio

print(f'{duracao.days} dia(s)')
print(f'{duracao.seconds // 3600} hora(s)')
duracao = duracao.seconds % 3600
print(f'{duracao // 60} minuto(s)')
duracao %= 60
print(f'{duracao} segundo(s)')

File: problemas/1000/1061.dart
import 'dart:io';

void main() {
  final diaInicio = int.parse(stdin.readLineSync().replaceAll('Dia ', ''));
  final horaInicio = stdin.readLineSync().replaceAll(' ', '');
  final diaFim = int.parse(stdin.readLineSync().replaceAll('Dia ', ''));
  final horaFim = stdin.readLineSync().replaceAll(' ', '');

  String doisDigitos(int n) {
    if (n >= 10) return '$n';
    return '0$n';
  }

  final dataInicio =
    DateTime.parse('${DateTime.now().year}-04-${doisDigitos(diaInicio)} $horaInicio');
  final dataFim =
    DateTime.parse('${DateTime.now().year}-04-${doisDigitos(diaFim)} $horaFim');
  final duracao = dataFim.difference(dataInicio);

  stdout.writeln('${duracao.inDays} dia(s)');
  stdout.writeln('${duracao.inHours.remainder(24)} hora(s)');
  stdout.writeln('${duracao.inMinutes.remainder(60)} minuto(s)');
  stdout.writeln('${duracao.inSeconds.remainder(60)} segundo(s)');
}